src/MongoDB.Driver.Core/Core/Clusters/MultiServerCluster.cs

```diff
@@ -417,11 +417,16 @@ private ClusterDescription ProcessReplicaSetChange(ClusterDescription clusterDes
                                     args.NewServerDescription.ReplicaSetConfig.Name,
                                     args.NewServerDescription.EndPoint,
                                     args.NewServerDescription.ReplicaSetConfig.Name)));
+
+                            _maxElectionInfo = new ElectionInfo(
+                                args.NewServerDescription.ReplicaSetConfig.Version.Value,
+                                args.NewServerDescription.ElectionId);
                         }
                         else
                         {
                             if (_maxElectionInfo.SetVersion < args.NewServerDescription.ReplicaSetConfig.Version.Value)
                             {
+                                var electionId = args.NewServerDescription.ElectionId ?? _maxElectionInfo.ElectionId;
```

Comment:
This change is required for behavior from spec ([link](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#actions)):

> ```
> if (description.setVersion is not null and
>     (topologyDescription.maxSetVersion is null or
>         description.setVersion > topologyDescription.maxSetVersion)):
> 
>     topologyDescription.maxSetVersion = description.setVersion
> ```
And is backed up by test ["Record max setVersion, even from primary without electionId"](https://github.com/mongodb/specifications/blob/5a7df21cbe7b1997483fe88eccfb2489eadf1d1a/source/server-discovery-and-monitoring/tests/rs/use_setversion_without_electionid.json#L2)

Reply:
Thanks for sharing the relevant version of the spec and the test, Mikalai!! That was super helpful.
I personally prefer reading YML to JSON, so in case other reviewers do as well, here's the [YML version of the above test](https://github.com/mongodb/specifications/blob/5a7df21cbe7b1997483fe88eccfb2489eadf1d1a/source/server-discovery-and-monitoring/tests/rs/use_setversion_without_electionid.yml).

Reply:
Just double checking: so the above logic simply states that when a node lacks an electionId, we should [keep the last electionId we received](https://github.com/mongodb/specifications/blob/5a7df21cbe7b1997483fe88eccfb2489eadf1d1a/source/server-discovery-and-monitoring/tests/rs/use_setversion_without_electionid.yml#L75)?

Reply:
Would it clearer to call `electionId`, `maxElectionId` then, since we're going end up saving it in `_maxElectionInfo`?

Reply:
`electionId` is just as clear to me since it reflects where the value came from (the `ElectionId`  in the server description.

Reply:
True. And the value  could also come from `_maxElectionInfo` I believe? I think this is a nit, so I defer to Mikalai on this :)

Reply:
> Just double checking: so the above logic simply states that when a node lacks an electionId, we should keep the last electionId we received?
Yes, it's the way it should work.
> Would it clearer to call electionId, maxElectionId then, since we're going end up saving it in _maxElectionInfo?
Thanks for suggestion, but I will go with existing `electionId` since Robert approves it.